ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/ApiIntegrationTest.java

```diff
@@ -659,6 +660,136 @@ private void shouldFailToExecuteQuery(final String sql, final String message) {
         startsWith(message));
   }
 
+  private void shouldFailToExecutePrint(final String sql, final String message) {
```

Comment:
Is this supposed to be a test? It's not annotated as such.

Reply:
No, this is used by two other tests "shouldFailToPrintNonExistingTopic" and "shouldFailToPrintWithInvalidSql".

Reply:
> Thanks @Guttz -- LGTM overall though the loss of type safety on the publisher type returned from the query endpoint is unfortunate. Given that you had to create new a new publisher, response writer, and test publisher for the new statement type, I think there's actually very little shared code between the existing query-handling path and the new print topic path, which is a good case for having a separate endpoint rather than piggybacking off the existing endpoint. I understand if you don't want to rewrite this PR with such a large change; I'm fine with it as-is though it is a bit unfortunate. Curious whether you agree / what you think the benefit of sharing the existing /query-stream endpoint is.

Thanks for the view, @vcrfxia! I agree with you that the code would seem cleaner and more modular if we had two endpoints here, and that the loss of a type is unfortunate. In other others, for our side of things it would be better if we separated it.

The reason I went in this direction was for a better developer experience when using our API. Since the print statement is mostly used for debugging or onboarding, I think it just feels more natural for the developer to be able to issue a print query to the same endpoint he just did for his push and pull queries and quickly get a result. Otherwise, he would probably first receive an error, google a bit and find out there's an extra /print endpoint. Also, we would have to modify some high level docs (i.e. https://docs.ksqldb.io/en/latest/developer-guide/api/) and add even more information and that only because of a statement which doesn't have a huge relevance. 

> Also, we'll need docs for this new feature as part of this PR before we merge.

Also, could you let me know which docs you mean exactly?